test/python/primitives/test_result.py

```diff
@@ -0,0 +1,74 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for BaseResult."""
+
+from dataclasses import dataclass
+from typing import Any, Collection
```

Comment:
If you use `from __future__ import annotations`, you can use `collections.abc.Collection` instead of `typing.Collection`.